### DISCUSSION TITLE
Should we expect changes that break compatibility?

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#1134] [Q&A] [answered]
I love the development of this app so far, and I want to migrate my library to it. 

However, I have an extensive library, and I only want to do it at the point where it is more or less sure that I won't have to do it again. What I am asking is basically what is the certainty that if I migrate all my pictures to immich now, there won't be some updates in the future that break something that would force me to re-add all the images. 

I understand this might be hard to predict, but I am looking for an approximate answer; on a scale of 0 to 10, is it a 1? a 5? or a 9?

**Comment by alextran1502** [comment, marked as answer]
I would say a 5. For example, when we implement the LivePhotos functionality. Files that are uploaded before that point doesn't have the motion part of live photos, if the user want to gain the functionality for all the assets, they will have to reupload those files from the mobile app.

**Reply by alextran1502** [reply]
Have the CLI help this significantly.

However, I don't see any potential features in the future that required a re-upload of your library

**Reply by jagjordi** [reply]
Thank you for the insight, I installed immich in my UNRAID setup a few days ago. Currently uploading my Library and really benefiting from the CLI. Great work!!
